### DISCUSSION TITLE
On invalidated event handler

### DISCUSSION THREAD

**Discussion by ThanhDat-Vo** [rrousselGit/riverpod#3734] [Ideas]
Look like there are no way to listen when a provider is invalidated for now.
Could we make one?
There is some event that need to be trigger when a provider is invalidated.

**Comment by noriHanda** [comment]
Is this what you are looking for? [ref.onDispose](https://pub.dev/documentation/riverpod/latest/riverpod/Ref/onDispose.html)

**Reply by ThanhDat-Vo** [reply]
@noriHanda well, sorry for not giving enough context in my question.
I knew about ref.onDispose already, but it only work inside provider itself.
Is there any way to listen outside of provider?
For example: 
```
abcProvider.onInvalidated((){
  debugPrint("abc is invalidated");
})
```

**Reply by noriHanda** [reply]
[ProviderObserver](https://pub.dev/documentation/riverpod/latest/riverpod/ProviderObserver-class.html) should do

**Reply by ThanhDat-Vo** [reply]
@noriHanda Thanks!
